src/get-script-to-run.js

```diff
@@ -10,19 +10,30 @@ function getScriptToRun(config, input) {
   // remove the default objects/strings so we cancheck
   // if the prefix works with another script first
   const defaultlessConfig = removeDefaults(cloneDeep(config))
-  const scriptString = getScriptString(defaultlessConfig, input)
-  if (scriptString) {
-    return scriptString
+  const scriptToRun = getScript(defaultlessConfig, input)
+  if (!isUndefined(scriptToRun) && isString(scriptToRun.script)) {
+    return scriptToRun
   } else {
     // fallback to the defaults if no other script was
     // found with the given input
-    return getScriptString(config, input)
+    return getScript(config, input)
   }
 }
 
-function getScriptString(config, input) {
+function getScript(config, input) {
   const [script] = prefixMatches(input, config)
-  return resolveScriptObjectToString(script)
+  if (script) {
+    const resolvedPrefix = Object.keys(script).shift()
+    let scriptToRun = script[resolvedPrefix]
+    if (resolvedPrefix && isPlainObject(scriptToRun)) {
+      scriptToRun = resolveScriptObjectToString(scriptToRun)
+    }
+    return {
+      resolvedPrefix,
```

Comment:
Could we call this `scriptName` instead? Because it's not the prefix, it's actually what the prefix was resolved to, which should be the full script name.